experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1373 @@
+    partition::Vector{Vector{Int}} = Vector{Int}[],
+    index_ordering_matrix::ZZMatrix = zero_matrix(ZZ, 0, 0)
+  )::DifferentialRanking{elem_type(base_ring(dpr))}
+  
+  __set_are_perms_up_to_date!(dpr, false)
+  m, n = nelementary_symbols(dpr), ndiffs(dpr)
+  dpr.ranking = DifferentialRanking{elem_type(base_ring(dpr))}(dpr, __compute_ranking_params(m, n, partition_name, index_ordering_name, partition, index_ordering_matrix)...)
+end
+
+function __compute_ranking_params(m::Int, n::Int,
+    partition_name,
+    index_ordering_name,
+    partition,
+    index_ordering_matrix
+  )::Tuple{Vector{Vector{Int64}}, ZZMatrix}
+  
+  @req partition_name in [:top, :pot, :default] "Invalid name of partition"
+  if partition_name == :default
+    if is_empty(partition)
+      partition = [fill(1, m)] #Use :top by default
+    else
+      # Otherwise the input is used. Check its validity:
+      @req __is_valid_partition(partition, m) "Not a partition of the number of elementary symbols"
+    end
+  else
+    if is_empty(partition)
+      if partition_name == :top
+        partition = [fill(1, m)]
+      else
+        partition = [[i == j ? 1 : 0 for j in 1:m] for i in 1:m]
+      end
+    else # This case is only accessed if both a partition and a name are provided. Then a consistency check is required.
+      @req __is_valid_partition(partition, m) "Not a partition of the number of elementary symbols"
+      if partition_name == :top
+        @req partition == [fill(1, m)] "The partition provided does not match its name"
+      else
+        @req partition == [[i == j ? 1 : 0 for j in 1:m] for i in 1:m] "The partition provided does not match its name"
+      end
+    end
+  end
+
+  @req index_ordering_name in [:default, :lex, :deglex, :invlex, :deginvlex, :degrevlex] "Invalid name of index ordering"
+  R, _ = polynomial_ring(QQ, n; cached = false) # We use a dummy polynomial ring to extract canonical matrices of monomial orderings
+  if index_ordering_name == :default
+    if is_empty(index_ordering_matrix)
+      index_ordering_matrix = canonical_matrix(lex(R)) #Use :lex by default
+    else
+      # Otherwise the input is used. Check its validity:
+      @req ncols(index_ordering_matrix) == n "The number of columns of the matrix provided must equal $n" 
+    end
+  else
+    @req is_empty(index_ordering_matrix) "Providing both a name and a matrix is not supported. Please just choose one."
+    #if is_empty(index_ordering_matrix)
+      if index_ordering_name == :lex
+        index_ordering_matrix = canonical_matrix(lex(R))
+      elseif index_ordering_name == :deglex
+        index_ordering_matrix = canonical_matrix(deglex(R))
+      elseif index_ordering_name == :invlex
+        index_ordering_matrix = canonical_matrix(invlex(R))
+      elseif index_ordering_name == :deginvlex
+        index_ordering_matrix = canonical_matrix(deginvlex(R))
+      elseif index_ordering_name == :degrevlex
+        index_ordering_matrix = canonical_matrix(degrevlex(R))
```

Comment:
I think you can get rid of all of this if-else business here by calling https://github.com/oscar-system/Oscar.jl/blob/449f7920ece3d85ee789226e1c96e8490a50845a/src/Rings/orderings.jl#L243 instead